### PR TITLE
Track setup events for sessions, errors, and logs

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

* Creates a new `setup_events` table for storing data about
* Starts populating the table with records when we process `MarkBackendSetup` events

The UI using this will come in another PR.

## How did you test this change?

Local click test.

## Are there any deployment considerations?

Will monitor to ensure events are created as expected.

### CODE REVIEW

File: backend/model/model.go [hunk 1]
```diff
@@ -335,6 +337,23 @@ type Project struct {
...
+	// Generic is temporary and can be removed once all messages are processed.
+	MarkBackendSetupTypeGeneric MarkBackendSetupType = "generic"
+	MarkBackendSetupTypeSession MarkBackendSetupType = "session"
+	MarkBackendSetupTypeError   MarkBackendSetupType = "error"
+	MarkBackendSetupTypeLogs    MarkBackendSetupType = "logs"
+)
+
+type SetupEvent struct {
+	ID        int                  `gorm:"primary_key;type:serial" json:"id" deep:"-"`
+	CreatedAt time.Time            `json:"created_at" deep:"-"`
```
Comment (ccschmitz) [inline backend/model/model.go:352]: I didn't use `Model` because we don't need `updated_at` or `deleted_at`.

**Review by Vadman97 — APPROVED** [review]
nice! just a comment about inserting the `SetupEvent` rows given he unique constraint

File: backend/otel/otel.go [hunk 2]
```diff
@@ -428,6 +431,7 @@ func (o *Handler) submitProjectLogs(ctx context.Context, projectLogs map[string]
 				Type: kafkaqueue.MarkBackendSetup,
 				MarkBackendSetup: &kafkaqueue.MarkBackendSetupArgs{
 					ProjectID: projectIDInt,
+					Type:      model2.MarkBackendSetupTypeLogs,
```
Comment (Vadman97) [inline backend/otel/otel.go:434]: @ccschmitz we should only record setup events here if this is not a frontend log. the `SubmitFrontendConsoleMessage` function adds a source attribute to differentiate it

Screenshot 2023-03-20 at 11 18 47 AM

Comment (ccschmitz) [reply backend/otel/otel.go:434]: Perfect - thanks, @Vadman97!

**Review by Vadman97 — APPROVED** [review]

**Pull request merged**